dart/dynamics/BodyNode.h

```diff
@@ -613,6 +591,60 @@ class BodyNode :
   template <class NodeType, typename ...Args>
   NodeType* createNode(Args&&... args);
 
+  /// Create an ShapeNode attached to this BodyNode. Pass an
+  /// ShapeNode::Properties argument into this function.
+  template <class ShapeNodeProperties>
+  ShapeNode* createShapeNode(const ShapeNodeProperties& properties);
+
+  /// Create an ShapeNode with the specified name
+  ShapeNode* createShapeNode(
+      const ShapePtr& shape, const std::string& name = "ShapeNode");
+
+  /// Create an ShapeNode with the specified name
+  ShapeNode* createShapeNode(const ShapePtr& shape, const char* name);
```

Comment:
As @mxgrey said, the implicit converting sometimes didn't work. I'm still not sure exactly when is the case. I just tested this function and `BodyNode::createEndEffector(const char*)`, and the implicit converting worked for both of them. So we can remove these functions.